src/components/selection_input/selection_input_store.ts

```diff
@@ -201,7 +207,7 @@ export class SelectionInputStore extends SpreadsheetStore {
    * e.g. ["A1", "Sheet2!B3", "E12"]
    */
   get selectionInputs(): (RangeInputValue & { isFocused: boolean; isValidRange: boolean })[] {
-    const generator = new ColorGenerator(this.colors);
+    const generator = new ColorGenerator(this.colors, getPaletteName(this.ranges.length));
```

Comment:
All the palette logic could be hidden in the ColorGenerator maybe ? With an argument colors + argument dataset length

Reply:
DO we ever want to do that ATM ?IF not, I think it's worth to simplify the use of the colorGenerator. We could always change it later if needed